task6/script.js

```diff
@@ -0,0 +1,508 @@
+   vendor: 'Буслик',
+   photoLink: 'https://buslik.by/upload/slam.image/iblock/a86/a86cec311e22d4396acf8b86e9f5efec-90.jpg',
+   validUntil: new Date('2021-05-10T23:00:00'),
+   discount: '50%',
+   hashTags: ['игрушки'],
+   rating: 5.0,
+   reviews :[]
+
+ },
+ {
+
+   id: '16',
+   description: 'Скидка на LEGO -15%',
+   createdAt: new Date('2021-03-15T23:00:00'),
+   link: 'https://buslik.by/catalog/igrushki_i_igry/konstruktory/lego/konstruktor_kosmicheskiy_robot_dlya_gornykh_rabot_327_detaley_lego_creator_31115/',
+   vendor: 'Буслик',
+   photoLink: 'https://buslik.by/upload/slam.image/iblock/ec4/ec4f0a6d7445e05aa2dd7bfb3a61306f-90.jpg',
+   validUntil: new Date('2021-03-30T23:00:00'),
+   discount: '15%',
+   hashTags: ['игрушки'],
+   rating: 5.0,
+   reviews :[]
+
+ },
+ {
+
+   id: '17',
+   description: 'Скидка на витамины -50%',
+   createdAt: new Date('2021-03-15T23:00:00'),
+   link: 'https://www.iherb.com/pr/California-Gold-Nutrition-Andrographis-Immune-with-AP-BIO-100-mg-120-Tablets/101059',
+   vendor: 'iHerb',
+   photoLink: 'https://s3.images-iherb.com/cgn/cgn01819/v/10.jpg',
+   validUntil: new Date('2021-04-15T23:00:00'),
+   discount: '50%',
+   hashTags: ['здоровье'],
+   rating: 4.0,
+   reviews :[]
+
+ },
+ {
+
+   id: '18',
+   description: 'Скидка на витамины -50%',
+   createdAt: new Date('2021-03-15T23:00:00'),
+   link: 'https://www.iherb.com/pr/Now-Foods-High-Potency-Vitamin-D-3-125-mcg-5-000-IU-120-Softgels/10421',
+   vendor: 'iHerb',
+   photoLink: 'https://s3.images-iherb.com/now/now00372/v/27.jpg',
+   validUntil: new Date('2021-05-04T23:00:00'),
+   discount: '50%',
+   hashTags: ['здоровье'],
+   rating: 5.0,
+   reviews :[]
+
+ },
+ {
+
+   id: '19',
+   description: 'Скидка на женские джинсы -36%',
+   createdAt: new Date('2021-03-15T23:00:00'),
+   link: 'https://www.wildberries.ru/catalog/15537164/detail.aspx?targetUrl=BP',
+   vendor: 'ТВОЕ',
+   photoLink: 'https://images.wbstatic.net/c516x688/new/15530000/15537164-1.jpg',
+   validUntil: new Date('2021-06-12T23:00:00'),
+   discount: '36%',
+   hashTags: ['одежда'],
+   rating: 4.0,
+   reviews :[]
+
+ },
+ {
+
+   id: '20',
+   description: 'Скидка на платье -12%',
+   createdAt: new Date('2021-03-15T23:00:00'),
+   link: 'https://www.wildberries.ru/catalog/19126078/detail.aspx?targetUrl=BP',
+   vendor: 'ТВОЕ',
+   photoLink: 'https://images.wbstatic.net/c516x688/new/19120000/19126078-1.jpg',
+   validUntil: new Date('2021-03-25T23:00:00'),
+   discount: '12%',
+   hashTags: ['одежда'],
+   rating: 5.0,
+   reviews :[]
+
+ },
+ ]
+
+ let getAds = (function(skip = 0, top = 10, filterConfig = {}){
+    let adsResult = [];
+    if (Object.keys(filterConfig).length !== 0 && filterConfig.constructor === Object) {
+        let key = Object.keys(filterConfig);
+        let value = filterConfig[key[0]];
+        if (key[0] === 'createdAt' || key[0] === 'validUntil') {
```

Comment:
Why do you check only first key? Maybe it's better to check all of them in loop?